pyccel/ast/builtin_methods/list_methods.py

```diff
@@ -83,7 +84,82 @@ def append_argument(self):
         """
         return self._append_arg
 
-#==============================================================================
+class ListInsert(PyccelInternalFunction):
+    """
+    Represents a call to the .insert() method.
+
+    Represents a call to the .insert() method of an object with a list type,
+    which inserts a given element at a given index in a list. This method returns `None`.
+    The insert method is called as follows:
+
+    >>> a = [2, 3, 4]
+    >>> a.insert(1, 1)
+    >>> print(a)
+    [1, 2, 3, 4]
+
+    Parameters
+    ----------
+    list_variable : Variable
+        The variable representing the list.
+
+    index : TypedAstNode
+        The index value for the element to be added.
+    
+    new_elem : TypedAstNode
+        The argument passed to insert() method.
+    """
+    __slots__ = ("_index", "_list_variable", "_insert_arg")
+    _attribute_nodes = ("_index", "_list_variable", "_insert_arg")
+    _dtype = NativeVoid()
+    _shape = None
+    _order = None
+    _rank = 0
+    _precision = -1
+    _class_type = NativeHomogeneousList()
```

Comment:
Insert returns a homogenous list of void objects?

Reply:
Certainly! Here's the corrected version:

Not really. Initially, it wasn't clear to me what the `_class_type` attribute was meant to represent. I thought it was related to the class that the method was associated with, but that was a stupid mistake. It should be `NativeVoid` as well.

Reply:
Not really. Initially, it wasn't clear to me what the `_class_type` attribute was meant to represent. I thought it was related to the class that the method was associated with, but that was a stupid mistake. It should be `NativeVoid` as well.

Reply:
fixed in b57fc5722e45d99a550c34b5d7db72559ce97fcd